### PR TITLE
Add test for utf-8 encoding of stdout/err

### PR DESCRIPTION
**Pull request by freakboy3742**
beeware/briefcase-macOS-Xcode-template#18 identified that macOS apps weren't being started with the correct system encoding. This PR adds a test case to verify this encoding.

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Comment by freakboy3742** [comment]
Two review notes: 

1. This PR extends the test-mode PR, so it shouldn't be merged until that PR is merged.
2. We have a chicken-and-egg problem with testing. This PR requires updated platform templates to pass CI; testing those templates requires this PR to add the test. At least for now, the templates aren't running test suites, but this is something we should consider.

File: tests/test_linux.py [hunk 1]
```diff
@@ -53,8 +53,8 @@ def test_gdk():
     gi.require_version("Gdk", "3.0")
     from gi.repository import Gdk
 
-    color = Gdk.Color(37, 42, 69)
-    assert color.to_string() == "#0025002a0045"
+    color = Gdk.RGBA(0.37, 0.42, 0.69, 0.5)
+    assert color.to_string() == "rgba(94,107,176,0.5)"
```
Comment (samschott) [inline tests/test_linux.py:57]: Did this change slip in accidentally or is it deliberate?

**Review by samschott — APPROVED** [review]

File: tests/test_linux.py (hunk 1 above)
Comment (freakboy3742) [reply tests/test_linux.py:57]: This was intentional; this test was causing a DeprecationWarning on Linux, so I took the opportunity to clean it up while I was in the code. It probably could/should have been a separate PR.

Comment (samschott) [reply tests/test_linux.py:57]: No worries, I was just confused.

**Review by mhsmith — APPROVED** [review]

**Pull request merged**